server/src/main/java/io/druid/server/http/InventoryViewUtils.java

```diff
@@ -37,42 +33,23 @@
 import java.util.Set;
 import java.util.TreeSet;
 
-public class InventoryViewUtils
+public interface InventoryViewUtils
 {
 
-  public static Set<DruidDataSource> getDataSources(InventoryView serverInventoryView)
+  static Set<ImmutableDruidDataSource> getDataSources(InventoryView serverInventoryView)
   {
-    TreeSet<DruidDataSource> dataSources = Sets.newTreeSet(
-        new Comparator<DruidDataSource>()
-        {
-          @Override
-          public int compare(DruidDataSource druidDataSource, DruidDataSource druidDataSource1)
-          {
-            return druidDataSource.getName().compareTo(druidDataSource1.getName());
-          }
-        }
-    );
-    dataSources.addAll(
-        Lists.newArrayList(
-            Iterables.concat(
-                Iterables.transform(
-                    serverInventoryView.getInventory(),
-                    new Function<DruidServer, Iterable<DruidDataSource>>()
-                    {
-                      @Override
-                      public Iterable<DruidDataSource> apply(DruidServer input)
-                      {
-                        return input.getDataSources();
-                      }
-                    }
-                )
-            )
-        )
-    );
-    return dataSources;
+    return serverInventoryView.getInventory()
+                              .stream()
+                              .flatMap(server -> server.getDataSources().stream())
+                              .map(DruidDataSource::toImmutableDruidDataSource)
+                              .collect(
```

Comment:
Why not just `toSet()`?

Reply:
Is it possible to collect as a TreeSet? This originally returns a TreeSet, and I don't want to change it.

Reply:
Could you please check this question, and leave a comment in code? as the method return type is just `Set`, not `NavigableSet` or `SortedSet`, it's suspicious

Reply:
Ah, I just realized this method is always called by the below ```getSecuredDataSources()``` and it converts the TreeSet to an ImmutableSet anyway. Changed to ```toSet()```.

Reply:
On ImmutableDruidDataSource, `idToSegments` hash code computation may take long

Reply:
Hmm, yes. In addition, a unit test expects the result is sorted by the name. I reverted this and changed the return type to SortedSet.